qiskit/quantum_info/operators/symplectic/pauli_list.py

```diff
@@ -177,19 +181,22 @@ def _from_paulis(data):
         if not isinstance(data, (list, tuple, set, np.ndarray)):
             data = [data]
         num_paulis = len(data)
-        if num_paulis == 0:
-            raise QiskitError("Input Pauli list is empty.")
         paulis = []
         for i in data:
             if not isinstance(i, Pauli):
                 paulis.append(Pauli(i))
             else:
                 paulis.append(i)
-        num_qubits = paulis[0].num_qubits
+        if num_qubits is None:
+            if num_paulis == 0:
+                raise QiskitError("Input Pauli list is empty; cannot infer num_qubits.")
+            num_qubits = paulis[0].num_qubits
         base_z = np.zeros((num_paulis, num_qubits), dtype=bool)
         base_x = np.zeros((num_paulis, num_qubits), dtype=bool)
         base_phase = np.zeros(num_paulis, dtype=int)
         for i, pauli in enumerate(paulis):
+            if pauli.num_qubits != num_qubits:
+                raise ValueError("num_qubits of Pauli does not match.")
```

Comment:
Would be nice to include values
```suggestion
                raise ValueError(f"num_qubits of Pauli does not match ({pauli.num_qubits} != {num_qubits}).")
```